src/library/Yi/Editor.hs

```diff
@@ -217,7 +217,7 @@ newRef :: MonadEditor m => m Int
 newRef = withEditor (refSupplyA %= (+ 1) >> use refSupplyA)
 
 newBufRef :: MonadEditor m => m BufferRef
-newBufRef = liftM BufferRef newRef
+newBufRef = fmap BufferRef newRef
```

Comment:
TLDR: AMP

The TravisCI failure is because

```haskell
class (Monad m, MonadState Editor m) => MonadEditor m
```

and as of GHC 7.10

```haskell
class Functor f => Applicative f
class Applicative m => Monad m
```

Hence `MonadEditor` implements `fmap` in GHC 7.10 because of the Functor context, but not in GHC 7.8

In theory, GHC 7.10 has the correct implementation. In practice, some workaround will be required.